Add limitations of using containers for developing hardware (#1672)
home-assistant/developers.home-assistant: docs/development_environment.mdx

The easiest way to get started with development is to use Visual Studio Code with devcontainers. This approach will create a preconfigured development environment with all the tools you need. This approach is enabled for all Home Assistant repositories. [Learn more about devcontainers.](https://code.visualstudio.com/docs/remote/containers)

> *Note:* as this approach uses containers, you may face challenges exposing hardware like USB-devices & adapters (on-board Bluetooth, Zigbee, ...) into the container for testing. This is in general possible when developing on a Linux-host, but you will not be able to directly access such hardware if you are using a Windows or MacOS computer for development.

Reviewer: ```suggestion
:::note
As this approach uses containers, you may face challenges exposing hardware like USB devices & adapters (onboard Bluetooth, Zigbee, ...) into the container for testing. This is possible when developing on a Linux host; however, you cannot directly access such hardware if you are using a Windows or MacOS computer for development.
:::
```
